src/main.zig

```diff
@@ -2462,8 +2460,20 @@ fn buildOutputType(
                     } else if (mem.eql(u8, "error", lookup_type)) {
                         linker_allow_shlib_undefined = false;
                     } else {
-                        fatal("unsupported -undefined option '{s}'", .{lookup_type});
+                        // There is an ambiguity here: ld on Linux uses `-(-)undefined <symbol>` to force a symbol to
+                        // be undefined during linking. Meanwhile, macOS ld uses it as the equivalent of Linux ld's
+                        // `-(-)(no-)allow-shlib-undefined` option.
+                        //
+                        // There are no great options here. For now, we allow the option to be used in both ways and
+                        // hope that no one is going to try to undefine a symbol called `dynamic_lookup` or `error`.
+                        //
+                        // Note that this ambiguity only exists for the linker option; the Clang option is explicitly
+                        // macOS-specific.
+                        try force_undefined_symbols.put(arena, lookup_type, {});
```

Comment:
The only other option I can think of here is to make the behavior dependent on `-target`. But that only works if `-target` comes first. I think it's also unprecedented for the option processing to be conditioned on target knowledge.

Reply:
In the face of ambiguity, refuse to proceed.

Reply:
What does that mean concretely here? If I follow that to the letter, I would have to rip out the `dynamic_lookup`/`error` handling that was already here (which I'm sure was added because someone needed it).

Reply:
@andrewrk ping just in case you missed the question above.

Reply:
I do think it should be followed to the letter, including ripping out the existing handling. However, I think it's possible to support unambiguously by lowering it as a placeholder that is resolved later when the target is known. Note this is for `zig cc` compatibility only; the regular zig CLI doesn't allow setting linker flags directly.

Reply:
Ok, I'll see if I can make that strategy work first, and remove it otherwise.